### PR TITLE
Raise error when tenant override fails

### PR DESCRIPTION
**Pull request by fuhuifang**
If an error occurs during tenant override, will not build and submit the pipeline.

### CODE REVIEW

File: shrike/pipeline/pipeline_helper.py [hunk 1]
```diff
@@ -1520,6 +1520,7 @@ def run(self):
                         )
                     except BaseException as e:
                         log.error(f"An error occurred, override is not successful: {e}")
+                        raise e
```
Comment (dkmiller) [inline shrike/pipeline/pipeline_helper.py:1523]: Just for my education, how is this related to #283 ?

Comment (fuhuifang) [reply shrike/pipeline/pipeline_helper.py:1523]: #283 handles the error during build and submit, this PR doesn't submit a pipeline if override fails. It's an issue brought up in https://stackoverflow.microsoft.com/questions/281579

**Review by jialei777 — APPROVED** [review]
